riocli/deployment/execute.py

```diff
@@ -22,44 +22,77 @@
 else:
     from riocli.utils.spinner import DummySpinner as Spinner
 
-from riocli.constants import Colors
-from riocli.deployment.util import select_details
-from riocli.utils.execute import run_on_cloud
-
+from riocli.config import new_v2_client
+from riocli.config import new_client
+from riocli.constants import Colors, Status
+from riocli.utils.execute import run_on_device
+from riocli.device.util import find_device_by_name
 
 @click.command(
     'execute',
     cls=HelpColorsCommand,
     help_headers_color=Colors.YELLOW,
     help_options_color=Colors.GREEN,
 )
-@click.option('--component', 'component_name', default=None,
-              help='Name of the component in the deployment')
+@click.option('--user', default='root')
+@click.option('--shell', default='/bin/bash')
 @click.option('--exec', 'exec_name', default=None,
               help='Name of a executable in the component')
 @click.argument('deployment-name', type=str)
 @click.argument('command', nargs=-1)
-# @name_to_guid
 def execute_command(
-        component_name: str,
+        user: str,
+        shell: str,
         exec_name: str,
         deployment_name: str,
-        deployment_guid: str,
         command: typing.List[str]
 ) -> None:
     """
-    Execute commands on cloud deployment
+    Execute commands on device deployment
     """
     try:
-        comp_id, exec_id, pod_name = select_details(deployment_guid, component_name, exec_name)
+        client = new_v2_client()
+
+        deployment = client.get_deployment(deployment_name)
+        if not deployment:
+            click.secho(f'Deployment `{deployment_name}` not found', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        if deployment.status.status != Status.RUNNING:
+            click.secho(f'Deployment `{deployment_name}` is not in Running state', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        if deployment.spec.runtime != 'device':
+            click.secho(f'Execution on cloud deployments is not supported', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        package = client.get_package(deployment.metadata.depends.nameOrGUID, query={"version": deployment.metadata.depends.version})
+        if exec_name is None:
+            executables = [e.name for e in package.spec.executables]
+            if len(executables) == 1:
+                exec_name = executables[0]
+            else:
+                exec_name = show_selection(executables, '\nChoose the executable')
+        
+        v1_client = new_client()
+        device = find_device_by_name(v1_client, deployment.spec.device.depends.nameOrGUID)
+
+        if not device:
```

Comment:
question: What will happen if the user provides an incorrect executable name?

If the error returned is not as intuitive, then it makes sense to also validate the user provided exec name against the ones in the package.

Reply:
error seems to be intuitive to me.

```
python3 -m riocli deployment execute testngindep   curl localhost  --exec romiq
Usage:
  dectl exec EXECUTABLE_NAME COMMAND [ARG...] [flags]

Flags:
  -h, --help   help for exec

Global Flags:
      --config string                   (default "/opt/rapyuta/device_edge/deviceedge.yaml")
      --log-flush-frequency duration   Maximum number of seconds between log flushes (default 5s)

2024-08-15 19:51:30.552806 I | finding executable: executable not found
```